compat/src/index.d.ts

```diff
@@ -1,75 +1,75 @@
 import * as _hooks from '../../hooks';
 import * as preact from '../../src';
 import { ForwardFn } from './internal';
-export * from './suspense';
-import { Suspense, lazy } from './suspense';
-
-export * from '../../hooks';
-export import Component = preact.Component;
-export import createContext = preact.createContext;
-export import createRef = preact.createRef;
-export import Fragment = preact.Fragment;
-export import createElement = preact.createElement
-export import cloneElement = preact.cloneElement
-
-export declare const version: string;
-
-export declare function createPortal(vnode: preact.VNode, container: Element | Element): preact.VNode<any>;
-
-export declare function render(vnode: preact.VNode<any>, parent: Element, callback?: () => void): Component | null;
-
-export declare function hydrate(vnode: preact.VNode<any>, parent: Element, callback?: () => void): Component | null;
-
-export declare function unmountComponentAtNode(container: Element | Document | ShadowRoot | DocumentFragment): boolean;
-
-export declare function createFactory(type: preact.VNode["type"]): preact.VNode<{}>;
-
-export declare function isValidElement(element: any): boolean;
-
-export declare function findDOMNode(component: preact.Component): Element | null;
-
-export declare interface PureComponent<P = {}, S = {}> extends preact.Component {
-  isPureReactComponenet: boolean;
+import * as _Suspense from './suspense';
+
+// export default React;
+export = React;
+declare namespace React {
+	// Export JSX
+	export import JSX = preact.JSX;
```

Comment:
This line currently causes `tsc` to crash. Removing it causes jsx to not be typed throwing errors.

Reply:
tsc crashes on this but we aren't seeing this in our tests or?

Reply:
Currently, JSX is not being exported in the compat typings (an oversight on my part in the initial PR). I added some specific tests for the typings, and running tsc with this line in will crash it, but removing the line causes `JSX.IntrinsicElements does not exist on JSX: any`.

Reply:
Should also note, that running `tsc` in `./tests/ts` also bails out with tsc crashing.

Reply:
Also, I modified an app from work to test this out, and it runs past this line with no problem (I use `awesome-typescript-loader` with webpack). On build it gives some different warnings:
```
WARNING in C:/Users/Peter/Documents/GitHub/preact/compat/dist/compat.module.js 1:5343-5344
"export 'Fragment' (imported as 'v') was not found in 'preact'
```

This may be build related as it doesn't seem TS related. Not sure.

Reply:
Fixed! After looking through the travis-ci logs I noticed that the typescript tests were passing, so it couldn't have been the TS version. Updated my global TS install to 3.5.1 and the issue is resolved!